var/spack/repos/builtin/packages/ffmpeg/package.py

```diff
@@ -0,0 +1,301 @@
+    version('0.8.2',   '967d481c98161582d149aced6e3b2f31')
+    version('0.8.1',   'e9c497e221b5594df477bcbf318f4bce')
+    version('0.8',     '7e9b8c8a6952de0c477027e48249f3ed')
+    version('0.7.17',  '5044faf253256c8129fa10ff11fdc642')
+    version('0.7.16',  '72eccaed0153d9ef57f0dbb7d797ca68')
+    version('0.7.15',  '4d502103002318c1d4c4d3475a2fe527')
+    version('0.7.14',  'c283472a18ae16bc8e49e1869f9a4ad6')
+    version('0.7.13',  'b226a3d4f81d45e4a81632f6e94b098c')
+    version('0.7.12',  'b4e3653b2dd930cfa2a80722f8b03182')
+    version('0.7.11',  'fb3de74e8a92152698aba429435ab73c')
+    version('0.7.10',  '61236188156d95f903aa0920019eea36')
+    version('0.7.9',   'e4d0c32c0a7cc763922ee8c34e688b5a')
+    version('0.7.8',   'e0053495f4cc7247600c4f2a3302dafb')
+    version('0.7.7',   'b3e268f01c3cea8fcbd4736a0d7f2806')
+    version('0.7.6',   'a9de4e2f077c4cd0434f426845478f14')
+    version('0.7.5',   '4e85037d1a94ff8cbcf5a0a133181a48')
+    version('0.7.4',   '96729126f805488739802118fdf06e75')
+    version('0.7.3',   'c9e455dc8f555028d551556798347abf')
+    version('0.7.2',   '79935897f2f910569235d15678ec99fc')
+    version('0.7.1',   '8615815985a17ef6498620c715e63f5e')
+    version('0.7',     'b215bd0a9ff82b74578264efdd955a08')
+    version('0.6.7',   '2afecd1ef31053ef077bdb2d6a7402ce')
+    version('0.6.6',   '904a4038eef339440ee0ffc494e51a25')
+    version('0.6.5',   'f8a49dea28bd33d3e5b12d1694ae8227')
+    version('0.6.4',   '945762d9a4c1a45aa395093b97d05ee9')
+    version('0.6.3',   'cdf4ad9b2a4d195b5ca874494bc7b0b0')
+    version('0.6.2',   'f7819307dad55aa0d6b40ac0e862884d')
+    version('0.6.1',   '4f5d732d25eedfb072251b5314ba2093')
+    version('0.6',     'd6142a9a5821d6a6262a6edb903faa24')
+    version('0.5.15',  '4019cbeed880af114190d353ff7d18ad')
+    version('0.5.14',  'bd713c6f79a4a1e18fd8833cc87002d9')
+    version('0.5.13',  '53f0815335f591e734faca010dcc354b')
+    version('0.5.12',  'c4abeeb9f030300467e569d533a9d80d')
+    version('0.5.11',  'b13a710fa3cb99cca17a31ff5907377e')
+    version('0.5.10',  '7d90176b5256cb8557922ea367662eb4')
+    version('0.5.9',   'f056fa753d84f0361ba35a5c44f43f6a')
+    version('0.5.8',   '570be94cce2a68c9efe5a8d7b338ff8e')
+    version('0.5.7',   'c50765544b84005eee65f722ddba3012')
+    version('0.5.6',   '4dc369e19df67f0e67a87a389a7dafbd')
+    version('0.5.5',   '9b1f676be005326a49a142b5a453f45d')
+    version('0.5.4',   '87e771cd0f5d465fbf1a0a4824b7cc24')
+    version('0.5.3',   '36c1045f013f6e8bac33a6200f1f554f')
+    version('0.5.2',   '451eb428ca97a72c00555d50944cdb24')
+    version('0.5.1',   'c7b0e1729f7aafb10496d79bb963bb26')
+    version('0.5',     'be8503f15c3b81ba00eb8379ca8dcf33')
+
+    variant('shared', default=False, description= \
+            'build shared libraries')
+    variant('pic', default=False, description= \
+            'build position-independent code')
+    variant('static', default=True, description= \
```

Comment:
we can, but it when someone uses this package, it will either be the default
`depends_on('foo')` which is the same as `depends_on('foo+shared')` or `depends_on('foo~shared')`. So you would still build two versions of `foo` if it's needed differently in different packages.